### PR TITLE
Signin error flow 322

### LINKED ISSUE #322: Create Account: Error flows
Update the error flows following Casey's designs: [Account Creation Specs]( https://www.figma.com/file/zwCNcZbWKB6uOBRTNXomYo/?node-id=1969%3A22536)

### PR DESCRIPTION
**Pull request by Carla-Moz**
This PR updates the sign-in error flow to match the designs in figma: [Account Creation](https://www.figma.com/file/zwCNcZbWKB6uOBRTNXomYo/?node-id=3%3A107)

See storybook for progress: [Storybook](https://storybook-rally.netlify.app/?path=/story/pages-signin--sign-in)

Extra small mobile size (320px):

image

closes #322

### CODE REVIEW

File: src/lib/components/auth-cards/CreateCard.svelte [hunk 1]
```diff
@@ -144,34 +215,31 @@
...
               on:change={handleChange}
               on:keyup={handleChange}
               id="id_user_email"
               name="id_user_email"
               type="email"
               width="100%"
               placeholder="Enter your email address"
-              required
             />
+            {#if createErr || emptyFieldsErr}
```
Comment (thomik-corp) [inline src/lib/components/auth-cards/CreateCard.svelte:228]: Can be simplified to if (emailErrText)

File: src/lib/components/auth-cards/CreateCard.svelte [hunk 2]
```diff
@@ -180,23 +248,27 @@
                 type="password"
                 {pattern}
                 width="100%"
-                required
```
Comment (thomik-corp) [inline src/lib/components/auth-cards/CreateCard.svelte:183]: Password should be required right?

File: src/lib/components/auth-cards/CreateCard.svelte [hunk 3]
```diff
@@ -205,7 +277,17 @@
               {/if}
             </div>
 
-            <ul class="info-rules">
+            {#if emptyFieldsErr || passwordErr}
```
Comment (thomik-corp) [inline src/lib/components/auth-cards/CreateCard.svelte:280]: Can change to if passwordErrText

File: src/lib/components/auth-cards/ForgetPWCard.svelte [hunk 4]
```diff
@@ -15,16 +15,18 @@
   export let height;
   export let store;
   export let sendUserInfo;
+  export let test;
 
-  let btnDisabled = true;
-  let email;
+  let custom = "";
```
Comment (thomik-corp) [inline src/lib/components/auth-cards/ForgetPWCard.svelte:20]: custom -> customClass?

File: src/lib/components/auth-cards/ForgetPWCard.svelte [hunk 5]
```diff
@@ -15,16 +15,18 @@
...
   let emailEl;
+  // let emptyFieldsErr;
```
Comment (thomik-corp) [inline src/lib/components/auth-cards/ForgetPWCard.svelte:22]: Can remove this

File: src/lib/components/auth-cards/LaunchCard.svelte [hunk 6]
```diff
@@ -18,11 +17,14 @@
   export let height;
   export let store;
   export let custom;
+  export let test = false;
```
Comment (thomik-corp) [inline src/lib/components/auth-cards/LaunchCard.svelte:20]: We can probably rename this to something more intuitive. I know this is existing code but still..

File: src/stories/components/LaunchCard.stories.svelte [hunk 7]
```diff
@@ -8,11 +8,14 @@
   import Button from "../../lib/components/Button.svelte";
   import Signin from "../../lib/components/auth-cards/SignInCard.svelte";
 
+  let handleTrigger = null;
+  let notVerified = false;
+  let notVerifiedText =
```
Comment (thomik-corp) [inline src/stories/components/LaunchCard.stories.svelte:13]: const?

File: src/lib/components/auth-cards/LaunchCard.svelte [hunk 8]
```diff
@@ -18,11 +17,14 @@
...
 
+  let notVerified = false;
+  let notVerifiedText =
```
Comment (thomik-corp) [inline src/lib/components/auth-cards/LaunchCard.svelte:23]: const?

File: src/lib/components/auth-cards/CreateCard.svelte [hunk 9]
```diff
@@ -18,95 +18,164 @@
...
+    }
+    return;
+  };
 
-      // Validate length
-      passwordEl.value.length >= minPasswordLength
-        ? length.classList.add("valid")
-        : length.classList.remove("valid");
+  const handleChange = (e) => {
+    const name = e.srcElement.name;
```
[the remark this answers is not in the document]
Comment (Carla-Moz) [reply src/lib/components/auth-cards/CreateCard.svelte:114]: Yes. Agreed.

File: src/lib/components/auth-cards/CreateCard.svelte [hunk 10]
```diff
@@ -18,95 +18,164 @@
...
 
+  const handleTrigger = (type) => {
+    dispatch("type", {
+      text: type,
+    });
+  };
+
+  const handleToggle = () => {
+    passwordVisible = !passwordVisible;
+    const type =
```
[the remark this answers is not in the document]
Comment (Carla-Moz) [reply src/lib/components/auth-cards/CreateCard.svelte:169]: Yes, true. Much better.

File: src/lib/components/auth-cards/SignInCard.svelte [hunk 11]
```diff
@@ -1,37 +1,98 @@
...
-  let email;
-  let password;
   let emailEl;
+  let emailErr = false;
+  let emailPattern = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/;
+  let emailErrText = null;
+  let emptyFieldsErr;
+  let fireBaseErr = null;
+  let inputClassName = "mzp-c-field-control";
+  let inputEmailName;
```
[the remark this answers is not in the document]
Comment (Carla-Moz) [reply src/lib/components/auth-cards/SignInCard.svelte:23]: Agreed. Done

File: src/lib/components/auth-cards/ForgetPWCard.svelte (hunk 4 above)
Comment (Carla-Moz) [reply src/lib/components/auth-cards/ForgetPWCard.svelte:20]: True. A lot clearer.

File: src/lib/components/auth-cards/LaunchCard.svelte (hunk 6 above)
Comment (Carla-Moz) [reply src/lib/components/auth-cards/LaunchCard.svelte:20]: Yes, this is for storybook testing. Agreed.

File: src/lib/components/auth-cards/ForgetPWCard.svelte (hunk 4 above)
Comment (Carla-Moz) [reply src/lib/components/auth-cards/ForgetPWCard.svelte:20]: Is that right? Nice, I'll check it out. Thanks.

**Pull request merged**